internal/orchestrator/orchestrator.go

```diff
@@ -357,41 +350,69 @@ func (or *orchestrator) AdminEvents() adminevents.Manager {
 	return or.adminEvents
 }
 
-func (or *orchestrator) initDatabaseCheckPreinit(ctx context.Context) (err error) {
-	if or.database == nil {
-		diType := config.GetString(coreconfig.DatabaseType)
-		if or.database, err = difactory.GetPlugin(ctx, diType); err != nil {
-			return err
+func (or *orchestrator) getDatabasePlugins(ctx context.Context) (plugins []database.Plugin, err error) {
+	dbConfigArraySize := databaseConfig.ArraySize()
+	plugins = make([]database.Plugin, dbConfigArraySize)
+	for i := 0; i < dbConfigArraySize; i++ {
+		config := databaseConfig.ArrayEntry(i)
+		name := config.GetString(database.DatabaseConfigName)
+		diType := config.GetString(database.DatabaseConfigType)
+		if name == "" || diType == "" {
+			return nil, i18n.NewError(ctx, coremsgs.MsgMissingDatabaseConfig)
 		}
-	}
-	if err = or.database.Init(ctx, databaseConfig.SubSection(or.database.Name()), or); err != nil {
-		return err
-	}
 
-	// Read configuration from DB and merge with existing config
-	var configRecords []*fftypes.ConfigRecord
-	filter := database.ConfigRecordQueryFactory.NewFilter(ctx).And()
-	if configRecords, _, err = or.GetConfigRecords(ctx, filter); err != nil {
-		return err
+		if err = core.ValidateFFNameField(ctx, name, "name"); err != nil {
+			return nil, err
+		}
+		plugin, err := difactory.GetPlugin(ctx, diType)
+		if err != nil {
+			return nil, err
+		}
+		plugins[i] = plugin
 	}
-	if len(configRecords) == 0 && config.GetBool(coreconfig.AdminPreinit) {
-		or.preInitMode = true
-		return nil
+
+	return plugins, err
+}
+
+func (or *orchestrator) initDatabasePlugins(ctx context.Context, plugins []database.Plugin) (err error) {
+	for idx, plugin := range plugins {
+		config := databaseConfig.ArrayEntry(idx)
+		err = plugin.Init(ctx, config.SubSection("sqlite3"), or)
```

Comment:
Feels like this should not be hard-coded to `sqlite3`

Reply:
oops that's an accident, thank you